src/test/components/elements/inputs/Password.test.tsx

```diff
@@ -0,0 +1,128 @@
+      PageComponent: Password,
+      componentProps: {
+        passwordFieldType: "New",
+        isPasswordVisible: false,
+        setIsPasswordVisible: setIsPasswordVisibleMock,
+        onChange: mockOnChange,
+        onBlur: mockOnBlur,
+        onFocus: mockOnFocus,
+        ...props,
+      },
+      initialRouteEntries: ["/register/student"],
+    });
+  };
+
+  const findPasswordInput = (type: "New" | "Confirm" | "Current") => {
+    const fieldType = passwordFieldTypes.find((field) => field.type === type);
+    const selector = `input[name="${fieldType?.expectedName}"]`;
+    return document.querySelector(selector) as HTMLInputElement;
+  };
+
+  const findToggleIcon = (type: "show" | "hide") => {
+    return screen.getByTestId(`${type}-password-icon`);
+  };
+
+  describe("Event Handling", () => {
+    test("handles onChange, onBlur, and onFocus events when typing", () => {
+      setupTest();
+      const passwordInput = findPasswordInput("New");
+      fireEvent.change(passwordInput, { target: { value: "newPassword123" } });
+      expect(mockOnChange).toHaveBeenCalledTimes(1);
+
+      fireEvent.blur(passwordInput);
+      expect(mockOnBlur).toHaveBeenCalledTimes(1);
+
+      fireEvent.focus(passwordInput);
+      expect(mockOnFocus).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("Toggle Icon", () => {
+    test("shows toggle icon and handles click, changing password to visible", () => {
+      setupTest({ showToggleIcon: true });
+      const passwordInput = findPasswordInput("New");
+      const toggleIcon = findToggleIcon("show");
+      expect(passwordInput).toHaveAttribute("type", "password");
+      fireEvent.click(toggleIcon);
+      expect(setIsPasswordVisibleMock).toHaveBeenCalledTimes(1);
+      expect(setIsPasswordVisibleMock).toHaveBeenCalledWith(true);
+    });
+
+    test("shows toggle icon and handles click, changing password to hidden", () => {
+      setupTest({
+        isPasswordVisible: true,
+        showToggleIcon: true,
+      });
+      const passwordInput = findPasswordInput("New");
+      expect(passwordInput).toHaveAttribute("type", "text");
+      const hideIcon = findToggleIcon("hide");
+      fireEvent.click(hideIcon);
+      expect(setIsPasswordVisibleMock).toHaveBeenCalledTimes(1);
+      expect(setIsPasswordVisibleMock).toHaveBeenCalledWith(false);
+    });
+
+    test("does not show toggle icon when 'showToggleIcon' prop is not provided", () => {
+      setupTest();
+      const toggleIcon = screen.queryByTestId("show-password-icon");
+      expect(toggleIcon).toBe(null);
+    });
+  });
+
+  describe("Rendering Styles", () => {
+    test("renders with invalid style when 'invalid' prop is true", () => {
+      setupTest({ invalid: true });
+      const passwordInput = findPasswordInput("New");
+      expect(passwordInput.classList.contains("is-invalid")).toBe(true);
+    });
+
+    test.each(passwordFieldTypes)(
+      "renders with correct name attribute for passwordFieldType: %s",
```

Comment:
The input is no longer a string, so this test title does not render as you'd expect. You can reword it to take both the type and the expected name into account.